iconservice/icx/issue/issue_regulator.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 ICON Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple
+
+from .regulator_storage import RegulatorStorage
+from ...database.db import ContextDatabase
+from ...iconscore.icon_score_context import IconScoreContext
+
+
+# todo: implement fee related logic
+class IssueRegulator:
+    _I_SCORE_EXCHANGE_RATE = 1_000
+
+    def __init__(self):
+        self._regulator_storage: 'RegulatorStorage' = None
+
+    def open(self, context_db: 'ContextDatabase'):
+        self.close()
+        self._regulator_storage = RegulatorStorage(context_db)
+
+    def close(self):
+        """Close resources
+        """
+        if self._regulator_storage:
+            self._regulator_storage.close(context=None)
+            self._regulator_storage = None
+
+    @staticmethod
+    def _reflect_difference_in_issuing(icx_issue_amount: int, over_issued_icx: int) -> Tuple[int, int, int]:
+        assert icx_issue_amount >= 0
+
+        corrected_icx_issue_amount = icx_issue_amount - over_issued_icx
+        if over_issued_icx < 0:
+            deducted_icx = over_issued_icx
+            remain_over_issued_icx = 0
+            return deducted_icx, remain_over_issued_icx, corrected_icx_issue_amount
+
+        if corrected_icx_issue_amount >= 0:
+            remain_over_issued_icx = 0
+            deducted_icx = over_issued_icx
+        else:
+            remain_over_issued_icx = abs(corrected_icx_issue_amount)
+            corrected_icx_issue_amount = 0
+            deducted_icx = icx_issue_amount
+
+        return deducted_icx, remain_over_issued_icx, corrected_icx_issue_amount
+
+    def _get_difference_between_icon_service_and_reward_calc(self, icx_amount, i_score_amount) -> int:
```

Comment:
I wonder why that method is existed.
because that code is enough to be inline ( 1line )

Reply:
I have removed this method